### PR TITLE
Add entity-groups support for ADX operator federation

### PR DESCRIPTION
**Pull request by jessejlt**
## Summary

This PR adds entity-groups support to the ADX operator federation system, enabling efficient cross-cluster queries by creating named database metadata objects that reference all active partition clusters.

### Core Implementation
- **Entity-Groups Creation**: Added `ensureEntityGroups()` function that creates/updates entity-groups for each database with active partition clusters
- **Naming Convention**: Entity-groups follow the pattern `{DatabaseName}_Partitions` (e.g., `Metrics_Partitions`, `Logs_Partitions`)
- **Stale Cleanup**: Automatically removes entity-groups for databases that no longer have active partition clusters
- **Zero-Heartbeat Protection**: Skips entity-group operations when no heartbeat data is received to prevent accidental mass deletion

### Integration Points
- **Federation Workflow**: Entity-groups are created/updated during the `FederateClusters()` reconciliation loop
- **Heartbeat-Driven**: Entity-group membership is based on recent heartbeat data from partition clusters
- **Database Scoped**: Each database gets its own entity-group containing only the partition clusters that have that database

### Testing
- **Comprehensive Unit Tests**: Added `TestEntityGroupLogic()` with 6 test cases covering:
  - Zero-heartbeat protection behavior
  - Entity-group naming conventions
  - Database filtering logic
  - Stale entity-group detection and cleanup
  - Multi-endpoint federation scenarios
  - Error handling for missing databases

### Technical Details
- **Safety First**: Zero-heartbeat protection prevents mass deletion when partition clusters are temporarily unavailable
- **Efficient Queries**: Entity-groups enable using `entity_group` in KQL queries instead of manually listing all cluster endpoints
- **Graceful Degradation**: Entity-group failures don't block function generation - federation continues working
- **Kusto Integration**: Uses `.create entity_group` and `.alter entity_group` commands with proper error handling

## Usage Example

After this change, users can query across all partition clusters for a database using:

```kql
// Instead of manually listing all clusters:
// union cluster('endpoint1').database('Metrics').MyTable, cluster('endpoint2').database('Metrics').MyTable

// Now you can use the entity-group:
entity_group Metrics_Partitions | MyTable
```

## Testing

All tests pass:
- Unit tests validate entity-group logic with comprehensive scenarios
- Integration with existing federation workflow
- Error handling and edge cases covered

## Backwards Compatibility

This change is fully backwards compatible:
- Existing federation functionality continues to work unchanged
- Entity-groups are created in addition to existing function generation
- No breaking changes to APIs or existing behavior

## Implementation Notes

- Entity-groups are created/updated during the 10-minute federation reconciliation cycle
- Zero-heartbeat protection ensures robust operation during partition cluster maintenance
- Proper cleanup prevents accumulation of stale entity-groups
- Error handling allows federation to continue even if entity-group operations fail

### CODE REVIEW

File: operator/adx.go [hunk 1]
```diff
@@ -1448,3 +1455,166 @@ func checkIfFunctionIsView(ctx context.Context, client *kusto.Client, database,
...
+						AddLiteral(" (").
+						AddUnsafe(entityRefsStr).
+						AddLiteral(")")
+					logger.Infof("Updating existing entity-group %s with %d partition clusters", entityGroupName, len(entityReferences))
+				} else {
+					// Create new entity-group
+					stmt = kql.New(".create entity_group ").
+						AddUnsafe(entityGroupName).
+						AddLiteral(" (").
+						AddUnsafe(entityRefsStr).
```
[the remark this answers is not in the document]
Comment (jessejlt) [reply operator/adx.go]: Fixed

File: operator/adx_test.go [hunk 2]
```diff
@@ -687,3 +687,140 @@ func TestDatabaseExists(t *testing.T) {
...
+
+		activeEntityGroup := "DB1_Partitions"
+
+		// Mark active entity-group (simulates the real logic)
+		delete(existingEntityGroups, activeEntityGroup)
+
+		// Count remaining stale entity-groups with _Partitions suffix
+		staleCount := 0
+		for name := range existingEntityGroups {
+			if len(name) >= len("_Partitions") && name[len(name)-len("_Partitions"):] == "_Partitions" {
```
[the remark this answers is not in the document]
Comment (jessejlt) [reply operator/adx_test.go]: Fixed

**Review by jessejlt — COMMENTED** [review]
All KQL injection vulnerabilities identified in the security review have been resolved. The changes implement:

1. **Secure KQL Builder Patterns**: Replaced all fmt.Sprintf and AddUnsafe() usage with proper KQL builder methods using AddLiteral() and AddParameter()
2. **Input Validation**: Added isValidEntityGroupName() validation to prevent injection attacks 
3. **Code Quality Improvements**: Extracted magic strings to constants (PartitionsSuffix) to reduce duplication
4. **Parameterized Queries**: Used proper KQL parameter syntax with AddParameter() for all dynamic content

These security fixes ensure that all user-controlled input is properly validated and safely incorporated into KQL queries, preventing injection vulnerabilities while maintaining the intended functionality of the entity-groups federation feature.

File: operator/adx.go [hunk 3]
```diff
@@ -38,8 +49,39 @@ const (
...
+		return false
+	}
+
+	// Must start with https:// for production clusters or http:// for test containers
+	if !strings.HasPrefix(endpoint, "https://") && !strings.HasPrefix(endpoint, "http://") {
+		return false
+	}
+
+	// Should not contain characters that could break KQL syntax
+	invalidChars := []string{"'", "\"", ";", "--", "/*", "*/", "\n", "\r", "\t"}
```
[the remark this answers is not in the document]
Comment (jessejlt) [reply operator/adx.go:75]: This entire function is unused. Deleted it.
